### PR TITLE
Scroll through sent chat messages with the arrow keys

### PR DESCRIPTION
**Pull request by Sturlen**
Use the up and down arrows to scroll through your sent chat messages/commands so you can easily repeat them. Inspired by the functionality in CMD and BASH.

example:
- type in and send message "/tp player"
- hit the up arrow key and enter to send the same command again.

Needs cleaning before it's merged, but I'd like to check in if anyone is working on something similar or for comments before spending too much time on it.

### CODE REVIEW

**Comment by RobbeBryssinck** [comment]
I'll ask Dragonisser to review for you

**Comment by Dragonisser** [comment]
Few things.

1. Might wanna limit the history to a certain size and do round-robin (E.g. only keep the last 50 messages, delete the first and append the new one)
2. No one works on it as of now even though it was internally discussed already
3. Has this been tested ingame?

**Comment by Sturlen** [comment]
Putting limits on history is probably a good idea.
Only been tested in the browser yet, but ingame testing is next on the todo list.

**Comment by Sturlen** [comment]
Tested in game, and both chat messages and commands work as intended.

**Review by Dragonisser — CHANGES_REQUESTED** [review]
Please change that part mentioned or comment why it wouldnt work. Then it can be merged.

File: Code/skyrim_ui/src/app/components/chat/message-history.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  private offset = 0;
+  /** When the history exeeds this number, the first item will be removed */
+  private maxHistoryLength: number;
+
+  constructor({maxHistoryLength}) {
+    this.maxHistoryLength = maxHistoryLength;
+  }
+
+  public push(newMsg: string) {
+    const isNewMessageDifferent = newMsg != this.stack[this.offset];
```
[the remark this answers is not in the document]
Comment (Sturlen) [reply Code/skyrim_ui/src/app/components/chat/message-history.ts:14]: I think it makes the code more readable, but that's just my preference. Variable has been removed.

**Review by Dragonisser — APPROVED** [review]
Can be merged.

**Pull request merged**